pkg/ast/sql/astsql.go

```diff
@@ -36,6 +37,33 @@ const (
 	Or
 )
 
+type Expression interface {
+	Evaluate() (interface{}, error)
+}
+
+var ErrInvalidArgumentCount = errors.New("invalid number of arguments")
+
+type SelectExprPlaceholder struct {
+	SelectExpr *sqlparser.SelectExpr
+}
+
+func NewSelectExprPlaceholder(selectExpr *sqlparser.SelectExpr) *SelectExprPlaceholder {
+	return &SelectExprPlaceholder{SelectExpr: selectExpr}
+}
+
+func (s *SelectExprPlaceholder) Evaluate() (interface{}, error) {
+	fmt.Println("Evaluating SelectExpr:", s.SelectExpr)
+	return nil, nil
+}
+
+type ExtractYearExpression struct {
+	Source Expression
+}
```

Comment:
Eventually we'll want to support the extract function in its entirety, so you should make this an `ExtractExpression` but for now you can just implement the functionality for extracting the year.